runtime/gc_modron_startup/mminit.cpp

```diff
@@ -2830,6 +2833,24 @@ gcInitializeDefaults(J9JavaVM* vm)
 	return J9VMDLLMAIN_FAILED;
 }
 
+static void
+adjustXmxXmsUsingRAMPercent(MM_GCExtensions *extensions, IDATA *memoryParameters)
+{
+	OMRPORT_ACCESS_FROM_OMRVM(extensions->getOmrVM());
+	uint64_t usableMemory = omrsysinfo_get_addressable_physical_memory();
```

Comment:
This function is called already in MM_GCExtensions::computeDefaultMaxHeap. In my opinion you should integrate handling of maxRAMPercent and initialRAMPercent in that code. Please not modify memoryParameters[opt_Xms] and memoryParameters[opt_Xmx]. Let them keep original command line option values if any.

Reply:
> In my opinion you should integrate handling of maxRAMPercent and initialRAMPercent in that code.

But computeDefaultMaxHeap() is called even before we parse gc options in gcParseCommandLineAndInitializeWithValues().

> Please not modify memoryParameters[opt_Xms] and memoryParameters[opt_Xmx]. 

Reason for modifying `memoryParameters[opt_Xms]` is the code in `gcInitializeCalculatedValues()` which sets `initialMemorySize` to default value if `memoryParameters[opt_Xms]` is not set.
So If I set `initialMemorySize` using initialRAMPercentage but don't update `memoryParameters[opt_Xms]`, it will get overwritten to default value in `gcInitializeCalculatedValues()`.

Reply:
Right, I see. Let me think about it a little bit.